### PR TITLE
ci: dart analyser

### LINKED ISSUE #671: feat: Automated Dart Analyzer Verification for Pull Request
### Type

Functionality

### Issue

Currently, there is no automated process in place to verify the Dart analyzer whenever a reviewer merges a pull request. This increases the risk of merging code that contains syntax or semantic errors, leading to potential bugs and decreased code quality.

### Feature

A feature request is submitted to implement an automated process that runs the Dart analyzer on the code in a pull request before it is merged. This would help ensure that the code is free of errors and maintain the high quality of the codebase.

### Motivation

Implementing this feature would improve the quality and reliability of the codebase by catching errors early in the development process. It would also save time for reviewers who would otherwise need to manually run the Dart analyzer on each pull request.

### Additional context

Related to #665

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] The issue is related solely to the ReVanced Manager

### PR DESCRIPTION
**Pull request by EvadeMaster**
## 🤖 - Dart analyser
Dart analyser to analyse Dart code quality, and will fail on **warning**

Resolved #671

### CODE REVIEW

**Review by PalmDevs — DISMISSED** [review]
Please see my review comments.

**Comment by PalmDevs** [comment]
Also if anything is related to CI, please always use the `ci` commit type. `refactor` is for refactoring **production code** changes only.

**Comment by PalmDevs** [comment]
You.. little shid. 💩

File: .github/workflows/analyze-code.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Analyze Code
+
+on:
+  push:
+    branches: [ "dev" ]
```
Comment (PalmDevs) [inline .github/workflows/analyze-code.yml:5]: Add `"main"` branch.

File: .github/workflows/analyze-code.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pull_request:
+    branches: [ "dev" ]
```
Comment (PalmDevs) [inline .github/workflows/analyze-code.yml:7]: Add `"main"` branch as well.

**Review by PalmDevs — CHANGES_REQUESTED** [review]
Please rename the workflow file to `analyze.yml` too, thanks.

File: .github/workflows/analyze-code.yml (hunk 1 above)
Comment (EvadeMaster) [reply .github/workflows/analyze-code.yml]: Resolved in fdd511

File: .github/workflows/analyze-code.yml (hunk 2 above)
Comment (EvadeMaster) [reply .github/workflows/analyze-code.yml]: Resolved in fdd511

**Review by PalmDevs — APPROVED** [review]
LGTM, wait for the changes to tested and then you can merge.

**Comment by Ushie** [comment]
> You.. little shid. 💩

Where are your manners? that's highly inappropriate.

**Comment by PalmDevs** [comment]
> > You.. little shid. 💩
> 
> Where are your manners? that's highly inappropriate.

**Pull request merged**